### DISCUSSION TITLE
How to assign the jobs with the pods which already exist?

### DISCUSSION THREAD

**Discussion by austinguish** [argoproj/argo-workflows#5423] [Q&A] [answered]
Hi, everyone.
I 'd like to ask if there anyway to create a workflow that uses the pods which already deployed without boot a new pod, or run `kubectl exec ` command in the cluster.
For example:
I have already set a kafka-client in the cluster. I want to create a workflow that produce some message. How to just set the exist kafka-client to produce without boot a new pod.
My idea is to use the template `resource` and set the action as 'exec' but it seems not work!
Thank u all !

**Comment by alexec** [comment, marked as answer]
A workflow cannot re-use existing pods. You can run `kubectl` as a step in a workflow using a [script template](https://argoproj.github.io/argo-workflows/walk-through/scripts-and-results/).

**Reply by austinguish** [reply]
Thank you !
